weather_report/weather_man.py

```diff
@@ -0,0 +1,102 @@
+"""weather_man.py
+   Has class that handle all data related functions
+"""
+import csv
+from typing import List, Dict, Tuple
+import os
+
+from weather_record import WeatherRecord
+
+
+class WeatherMan:
+    """Class responsible for handling all functions on weather data
+    """
+
+    _weather_records: List[WeatherRecord] = []
+
+    def __init__(self) -> None:
+        if not WeatherMan._weather_records:
+            weather_data = self._get_folder_data()
+            for record in weather_data:
+                WeatherMan._weather_records.append(WeatherRecord(record))
+
+    def _get_folder_data(self) -> List[Dict[str, str]]:
+        weather_data_dir = os.path.join(
+                                        os.path.dirname(__file__),
+                                        'weatherfiles'
+                                        )
+        data_files = list(os.listdir(weather_data_dir))
+        data = []
+        for file_path in data_files:
+            with open(
+                os.path.join(weather_data_dir, file_path),
+                "r", encoding="latin-1"
+            ) as f:
+                csv_reader = csv.DictReader(x.replace("\0", "") for x in f)
+                data.extend(list(csv_reader))
+        return data
+
+    def get_month_data(self, year: int, month: int) -> List[WeatherRecord]:
+        """Return data of a particular month of a year
+        """
+        month_data = []
+        for record in self._weather_records:
+            if (record.date.year, record.date.month) == (year, month):
+                month_data.append(record)
+        return month_data
```

Comment:
```
def filter_data(self, target_date: datetime):
```
This will filter weather data according to provided date time. You can use it in filtering data for any command.